### PR TITLE
Cleanup point-related code

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: None```

#### Purpose of change

Some cleanup to point-related code.

#### Describe the solution

See commit messages.

### CODE REVIEW

File: src/point.h [hunk 1]
```diff
@@ -349,6 +349,19 @@ static constexpr std::array<point, 4> four_adjacent_offsets{{
         point_north, point_east, point_south, point_west
     }};
 
+static constexpr std::array<point, 4> neighborhood{ {
+        point_south, point_east, point_west, point_north
+    }};
+
+static constexpr std::array<point, 4> offsets = {{
```
Comment (BevapDin) [inline src/point.h]: This contains the same values as `four_adjacent_offsets` and the name does not give an indication why one could not simply use the other array and what exactly the difference is. Is this array even needed?

In fact, even `neighborhood` contains the same thing. How is any reader supposed to know what the difference is (the order of the contained points)?

File: src/vehicle.cpp [hunk 2]
```diff
@@ -1681,8 +1679,8 @@ bool vehicle::is_connected( const vehicle_part &to, const vehicle_part &from,
         discovered.pop_front();
         auto current = current_part.mount;
 
-        for( int i = 0; i < 4; i++ ) {
-            point next = current + vehicles::cardinal_d[i];
+        for( const point &cardinal_direction : five_cardinal_directions ) {
```
Comment (BevapDin) [inline src/vehicle.cpp]: The original loop only goes through the first 4 directions. (It stops after `i == 3`, the loop body is not run when `i == 4`). Applies to the loops below as well.

File: src/point.h (hunk 1 above)
Comment (ZhilkinSerg) [reply src/point.h]: While container contents are the same, the order of elements is different (and that matters in some contexts). I am trying to get rid of duplicate code and moving all similar collections to one place is a first step. Ultimately I want to use a single collection.

File: src/vehicle.cpp (hunk 2 above)
Comment (ZhilkinSerg) [reply src/vehicle.cpp]: Thanks. I missed that - I should use a different point container (w/o `point_zero`) here and below then.

Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
        for( const point &offset : four_adjacent_offsets ) {
```

File: src/vehicle.cpp [hunk 3]
```diff
@@ -1681,8 +1679,8 @@ bool vehicle::is_connected( const vehicle_part &to, const vehicle_part &from,
...
+            point next = current + cardinal_direction;
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
            point next = current + offset;
```

File: src/vehicle.cpp [hunk 4]
```diff
@@ -1897,18 +1897,19 @@ bool vehicle::merge_rackable_vehicle( vehicle *carry_veh, const std::vector<int>
...
                     break;
                 }
+                ++j;
             }
 
             // We checked the adjacent points from the mounting rack and managed
             // to find the current structure part were looking for nearby. If the part was not
             // near this particular rack, we would look at each in the list of rack_parts
-            const bool carry_part_next_to_this_rack = j < 4;
+            const bool carry_part_next_to_this_rack = j < five_cardinal_directions.size() - 1;
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
            const bool carry_part_next_to_this_rack = j < four_adjacent_offsets.size();
```

File: src/vehicle.cpp [hunk 5]
```diff
@@ -2289,8 +2290,8 @@ bool vehicle::find_and_split_vehicles( int exclude )
                 veh_parts.push_back( p );
             }
             checked_parts.insert( test_part );
-            for( size_t i = 0; i < 4; i++ ) {
-                const point dp = parts[test_part].mount + vehicles::cardinal_d[ i ];
+            for( const point &cardinal_direction : five_cardinal_directions ) {
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
            for( const point &offset : four_adjacent_offsets ) {
```

File: src/vehicle.cpp [hunk 6]
```diff
@@ -2289,8 +2290,8 @@ bool vehicle::find_and_split_vehicles( int exclude )
...
+                const point dp = parts[test_part].mount + cardinal_direction;
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
                const point dp = parts[test_part].mount + offset;
```

File: src/vehicle.cpp [hunk 7]
```diff
@@ -5645,11 +5646,14 @@ void vehicle::refresh_insides()
             continue;
         }
 
-        parts[p].inside = true; // inside if not otherwise
-        for( int i = 0; i < 4; i++ ) { // let's check four neighbor parts
-            point near_mount = parts[ p ].mount + vehicles::cardinal_d[ i ];
+        // inside if not otherwise
+        parts[p].inside = true;
+        // let's check four neighbor parts
+        for( const point &cardinal_direction : five_cardinal_directions ) {
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
        for( const point &offset : four_adjacent_offsets ) {
```

File: src/vehicle.cpp [hunk 8]
```diff
@@ -5645,11 +5646,14 @@ void vehicle::refresh_insides()
...
+            point near_mount = parts[ p ].mount + cardinal_direction;
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
            point near_mount = parts[ p ].mount + offset;
```

File: src/point.h [hunk 9]
```diff
@@ -349,6 +349,19 @@ static constexpr std::array<point, 4> four_adjacent_offsets{{
...
+        point_south, point_east, point_west, point_north
+    }
+};
+
```
Comment (ZhilkinSerg) [inline src/point.h]:
```suggestion

static constexpr std::array<point, 4> four_cardinal_directions{{
        point_west, point_east, point_north, point_south
    }};

```

File: src/vehicle.cpp [hunk 10]
```diff
@@ -1897,18 +1897,19 @@ bool vehicle::merge_rackable_vehicle( vehicle *carry_veh, const std::vector<int>
             // There's no mathematical transform from global pos3 to vehicle mount, so search for the
             // carry part in global pos3 after translating
             point carry_mount;
-            for( j = 0; j < 4; j++ ) {
-                carry_mount = parts[ rack_part ].mount + vehicles::cardinal_d[ j ];
+            for( const point &cardinal_direction : five_cardinal_directions ) {
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
            for( const point &offset : four_cardinal_directions ) {
```

File: src/vehicle.cpp [hunk 11]
```diff
@@ -1897,18 +1897,19 @@ bool vehicle::merge_rackable_vehicle( vehicle *carry_veh, const std::vector<int>
...
+                carry_mount = parts[ rack_part ].mount + cardinal_direction;
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
                carry_mount = parts[ rack_part ].mount + offset;
```

File: src/vehicle.cpp [hunk 12]
```diff
@@ -1812,8 +1810,9 @@ bool vehicle::try_to_rack_nearby_vehicle( const std::vector<std::vector<int>> &l
         partial_matches.assign( 4, veh_partial_match );
         for( auto rack_part : this_bike_rack ) {
             tripoint rack_pos = global_part_pos3( rack_part );
-            for( int i = 0; i < 4; i++ ) {
-                tripoint search_pos( rack_pos + vehicles::cardinal_d[ i ] );
+            int i = 0;
+            for( const point &cardinal_direction : five_cardinal_directions ) {
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
            for( const point &offset : four_cardinal_directions ) {
```

File: src/vehicle.cpp [hunk 13]
```diff
@@ -1812,8 +1810,9 @@ bool vehicle::try_to_rack_nearby_vehicle( const std::vector<std::vector<int>> &l
...
+                tripoint search_pos( rack_pos + cardinal_direction );
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]:
```suggestion
                tripoint search_pos( rack_pos + offset );
```

**Pull request merged**